server/src/controllers/job_lead/getAllJobLeads.js

```diff
@@ -54,6 +56,21 @@ const getAllJobLeadsRequestHandler = async (req, res) => {
       };
     }
 
+    let employerIds = [];
+    if (searchEmployerNameQuery) {
+      const employers = await Employer.findAll({
+        where: {
+          name: { [Op.like]: `%${searchEmployerNameQuery}%` },
+        },
+        attributes: ["id"],
+      });
+      employerIds = employers.map((employer) => employer.id);
+    }
+
+    if (employerIds.length > 0) {
+      query.employer = { [Op.in]: employerIds };
+    }
```

Comment:
I think we should move this code block inside of the "`if (searchEmployerNameQuery)`" statement. Alongside this change I think we should remove the "`if (employerIds.length > 0)`" check. The result of applying these two changes is that whenever "searchEmployerNameQuery" is set, we only return the employers with the given name.
In the current implementation if no employer matches the given name, we return all of them.

Reply:
Ok got that fixed up